### PR TITLE
Use PollingBlockNotifier with MetaMask

### PR DESCRIPTION
**Pull request by tinybike**
Fixes https://app.clubhouse.io/augur/story/9797/new-blocks-not-appearing-in-ui-when-using-metamask

### CODE REVIEW

File: src/block-management/block-notifier.js [hunk 1]
```diff
@@ -3,18 +3,21 @@
...
-    blockNotifier.subscribe(this.notifySubscribers);
-  }.bind(this));
+  } else {
+    blockNotifier = new SubscribingBlockNotifier(transport, function () {
+      blockNotifier.destroy();
+      blockNotifier = new PollingBlockNotifier(transport, pollingIntervalMilliseconds);
+      blockNotifier.subscribe(this.notifySubscribers);
+    }.bind(this));
+  }
   blockNotifier.subscribe(this.notifySubscribers);
```
Comment (epheph) [inline src/block-management/block-notifier.js]: Will this double-subscribe if it goes via the non-metamask but failed subscribe path?

File: src/utils/is-meta-mask.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+"use strict";
+
+function isMetaMask() {
```
Comment (epheph) [inline src/utils/is-meta-mask.js]: Use the same function here? https://github.com/ethereumjs/ethrpc/blob/7f63d147bdec1e17bab7618d28ed22f464c5ee6c/src/transport/web3-transport.js#L15

File: src/block-management/block-notifier.js (hunk 1 above)
Comment (tinybike) [reply src/block-management/block-notifier.js]: Yep, but the first block notifier is destroyed, so it should be fine.  (It worked this way before, too.)

Comment (epheph) [reply src/block-management/block-notifier.js]: oh i see, yes, happens later  in the callback 👍

File: src/utils/is-meta-mask.js (hunk 2 above)
Comment (tinybike) [reply src/utils/is-meta-mask.js]: It's not quite the same function, but yes there's some duplicate code, added a global web3 check function.

**Pull request merged**